indexer/src/main/scala/scalding/FeatureEditHelper.scala

```diff
@@ -0,0 +1,276 @@
+// Copyright 2014 Foursquare Labs Inc. All Rights Reserved.
+package com.foursquare.twofishes.scalding
+
+import com.foursquare.twofishes._
+import java.nio.ByteBuffer
+import com.foursquare.twofishes.util.NameNormalizer
+import com.vividsolutions.jts.io.{WKBWriter, WKTReader}
+import org.geotools.geojson.geom.GeometryJSON
+
+object FeatureEditHelper {
+
+  // TODO(rahul): refactor to share code with ConcreteHotfixStorageService
+
+  def applyEdits(feature: GeocodeServingFeature, edits: GeocodeServingFeatureEdits): Option[GeocodeServingFeature] = {
+    applyEdits(feature, edits.edits)
+  }
+
+  def applyEdits(feature: GeocodeServingFeature, edits: Seq[GeocodeServingFeatureEdit]): Option[GeocodeServingFeature] = {
+    // if there is a Remove edit, return None right away
+    if (edits.exists(edit => edit.editType == EditType.Remove)) {
+      None
+    } else {
+      var featureMutable = feature.mutableCopy
+      for {
+        edit <- edits
+        if edit.editType == EditType.Modify
+      } {
+        featureMutable = applyModifyEdit(featureMutable, edit)
+      }
+      Some(featureMutable)
+    }
+  }
+
+  private val wktReader = new WKTReader()
+  private val wkbWriter = new WKBWriter()
+  private val geometryJSON = new GeometryJSON()
+
+  private def processLongListEdits(list: Seq[Long], edits: Seq[LongListEdit]): Seq[Long] = {
+    var listCopy = list
+    edits.foreach(edit => {
+      edit.editType match {
+        case EditType.Add => listCopy = listCopy.filterNot(_ == edit.value) :+ edit.value
+        case EditType.Remove => listCopy = listCopy.filterNot(_ == edit.value)
+        case _ =>
+      }
+    })
+    listCopy
+  }
+
+  private def processStringListEdits(list: Seq[String], edits: Seq[StringListEdit]): Seq[String] = {
+    var listCopy = list
+    edits.foreach(edit => {
+      edit.editType match {
+        case EditType.Add => listCopy = listCopy.filterNot(_ == edit.value) :+ edit.value
+        case EditType.Remove => listCopy = listCopy.filterNot(_ == edit.value)
+        case _ =>
+      }
+    })
+    listCopy
+  }
+
+  private def processFeatureNameFlagsListEdits(list: Seq[FeatureNameFlags], edits: Seq[FeatureNameFlagsListEdit]): Seq[FeatureNameFlags] = {
+    var listCopy = list
+    edits.foreach(edit => {
+      edit.editType match {
+        case EditType.Add => listCopy = listCopy.filterNot(_ == edit.value) :+ edit.value
+        case EditType.Remove => listCopy = listCopy.filterNot(_ == edit.value)
+        case _ =>
+      }
+    })
+    listCopy
+  }
+
+  def processFeatureNameListEdits(list: Seq[FeatureName], edits: Seq[FeatureNameListEdit]): Seq[FeatureName] = {
+    var listCopy = list
```

Comment:
likewise here you could do this without mutation though I admit it may be a bit of a pain